Tests/Shared.Specs/ComparableSpecs.cs

```diff
@@ -538,6 +571,43 @@ public override string ToString()
         }
     }
 
+    public class SameInstanceIsNotEqualClass
```

Comment:
I would keep these helper classes right after the last test method that uses it. We did this differently in the past, but I think this aligns closer to the "reads like a book" from Clean Code.